File: Flutter Frontend/smart_class/lib/models/location_model.dart
class LocationModel{
  String name;
  double latitude;
  String?  id;
  double longitude;
  String teacher;
  LocationModel({
    required this.name,
    required this.longitude,
    this.id,
    required this.latitude,
    required this.teacher,
});

  Map<String, dynamic> toJson()
  {
    return {
      "name" : name,
      "longitude" : longitude,
      "_id" : id,
      "latitude" : latitude,
      "teacher" : teacher,
    };
  }

factory LocationModel.fromJson(Map<String, dynamic> json){
    return LocationModel(name: json['name'], longitude: json['longitude'], latitude: json['latitude'], teacher: json['teacher'], id: json['_id']);
}
}

File: Backend/models/location_model.js
const mongoose = require("mongoose");

const locationModel = mongoose.Schema({
  name: String,
  latitude: Number,
  longitude: Number,
  teacher: {
    type: mongoose.Schema.Types.ObjectId,
    ref: "Teachers",
  },
});

module.exports = mongoose.model("Locations", locationModel);
